### PR TITLE
OnBoarding: Add support for xdist

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
## Motivation

We want to run the onboarding tests in parallel after the scenario/aws provision is started

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -103,8 +103,8 @@ def pytest_sessionstart(session):
 
     if session.config.option.collectonly:
         return
-
-    context.scenario.session_start()
+    if not hasattr(session.config, "workerinput"):
+        context.scenario.session_start()
```
Comment (robertomonteromiguel) [inline conftest.py:107]: Too risky. My case works!!! ...but the other cases...

Comment (cbeauchesne) [reply conftest.py:107]: gooooood catch !

File: requirements.txt [hunk 2]
```diff
@@ -4,6 +4,7 @@ pytest==7.1.3
 pytest-json-report==1.5.0
 pytest-split==0.9.0
 pytest-xdist==3.2.1
+pluggy==1.0.0
```
Comment (robertomonteromiguel) [inline requirements.txt:7]: I don't know why, in my local I'm downloading the 1.0.0 version, but in the gitlab when I execute 
"./build.sh -i runner" it's installing version "pluggy-1.5.0"
This causes failures:
`/go/src/github.com/DataDog/system-tests/venv/lib/python3.9/site-packages/_pytest/main.py:303: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: 1386284331[458](https://gitlab.ddbuild.io/DataDog/system-tests/-/jobs/626652661#L458)08, Hook: pytest_sessionfinish
OSError: cannot send (already closed?)
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config.hook.pytest_sessionfinish(`

File: run.sh [hunk 3]
```diff
@@ -418,6 +418,9 @@ function main() {
         if [[ "${scenario}" == K8S_LIBRARY_INJECTION_* ]]; then
             pytest_numprocesses=$(nproc)
         fi
+        if [[ "${scenario}" == *_AUTO_INJECTION ]]; then
+            pytest_numprocesses=6
```
Comment (robertomonteromiguel) [inline run.sh:422]: Avoid problems with gitlab kubernetes. xdist is resolving all cpu of the kubernetes machine, not the number of cpus allowed by pod

**Pull request merged**